kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java

```diff
@@ -201,8 +210,13 @@ public void start(Map<String, String> props) {
     }
   }
 
-  private SourceRecord createRecordWithHeaders(Map<String, String> messageAttributes, Map<String,String> ack,
-                                               String key, byte[] messageBytes, Long timestamp) {
+  private SourceRecord createRecordWithHeaders(
+      Map<String, String> messageAttributes,
+      Map<String, String> ack,
+      String key,
+      String orderingKey,
+      byte[] messageBytes,
+      Long timestamp) {
     ConnectHeaders headers = new ConnectHeaders();
     for (Entry<String, String> attribute :
             messageAttributes.entrySet()) {
```

Comment:
Maybe we should also add the ordering key as an attribute to the message sent to Kafka? Independent of whether we use it to partition or not.

Reply:
Yes, good idea. I have added it in a single commit added to this PR if you want to take a look: https://github.com/GoogleCloudPlatform/pubsub/pull/240/commits/07d5c3d0efbac1cd1d512ba006dc872933f6fe7d.